backend/tests/integration/routers/test_idioms.py

```diff
@@ -67,6 +67,137 @@ def test_get_idioms_sort_imagery(test_server, idioms_test_data):
     assert_idioms(actual_idioms, expected_idioms)
 
 
+def test_get_categories_returns_all_unique_categories(test_server, idioms_test_data):
+    response = test_server.get("/idioms/categories")
+    assert response.status_code == 200
+
+    categories = response.json()
+    assert isinstance(categories, list)
+    assert len(categories) > 0
+
+    # Collect all expected categories from test data
+    expected_categories = set()
+    for idiom in idioms_test_data.values():
+        expected_categories.update(idiom.context_diversity)
```

Comment:
this can be `expected_categories = sorted(set(idioms_test_data.values))`

which takes care of the sorting, you might need to cast to list first though, afaik sets are not sortable, might be that that changed in newer python versions, did you try running the tests? `uv run pytest` should do it